src/services/__tests__/osidb-errors-helpers.spec.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  getDisplayedOsidbError,
+  parseOsidbErrors,
+} from '../osidb-errors-helpers';
+
+
+describe('getDisplayedOsidbError', () => {
+
+  it('should return a string with the error message', () => {
+    const error = {
+      response: {
+        status: 500,
+        statusText: 'Internal Server Error',
+        data: 'Internal Server Error',
+      },
+    };
+    expect(getDisplayedOsidbError(error)).toBe(
+      'OSIDB responded with error 500 (Internal Server Error). \nInternal Server Error'
+    );
+  });
+
+  it('should return a string with the error message when error.response.data is an object', () => {
+    const error = {
+      response: {
+        status: 500,
+        statusText: 'Internal Server Error',
+        data: { error: 'Internal Server Error' },
+      },
+    };
+    expect(getDisplayedOsidbError(error)).toBe(
+      'OSIDB responded with error 500 (Internal Server Error). \nerror: Internal Server Error'
+    );
+  });
+
+  it('should return a string with the error message when error.request is an object', () => {
+    const error = {
+      request: { error: 'Internal Server Error' },
+    };
+    const expectJsonStringified = '{\n  "error": "Internal Server Error"\n}';
+    expect(getDisplayedOsidbError(error)).toBe(expectJsonStringified);
+  });
+
+  it('should return a string with the error message when error.request is **NOT** an object', () => {
```

Comment:
Not really a suggestion, just wanted to expose it in case you don't know.
I think that for this kind of tests, **tableDriven tests** ([vitest docs](https://vitest.dev/api/#test-each)) make a lot of sense, for example:

```js
it.each([
  {
    error: {/* omitted for simplicity */ },
    expected: 'OSIDB responded with error 500 (Internal Server Error). \nInternal Server Error',
    return: 'error.response.data is a string'
  },
  {
    error: {/* omitted for simplicity */ },
    expected: 'OSIDB responded with error 500 (Internal Server Error). \nerror: Internal Server Error',
    return: 'error.response.data is an object'
  },
  {
    error: {/* omitted for simplicity */ },
    expected: '{\n "error": "Internal Server Error"\n}',
    return: 'error.request is an object'
  }
])('should return a string with the error message when $return', ({ error, expected }) => {
  expect(getDisplayedOsidbError(error)).toBe(expected)
})
```

Reply:
This is great, thanks for sharing the knowledge!